### PR TITLE
ejdb: Build a bottle for Linuxbrew

### PR DESCRIPTION
**Pull request by jonchang**

### CODE REVIEW

**Comment by jonchang** [comment]
https://travis-ci.org/Linuxbrew/homebrew-core/builds/213722004#L2569

```
./test: error while loading shared libraries: libbz2.so.1.0: cannot open shared object file: No such file or directory
```

File: Formula/ejdb.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@ class Ejdb < Formula
   end
 
   depends_on "cmake" => :build
+  depends_on "bzip" unless OS.mac?
```
Comment (sjackman) [inline Formula/ejdb.rb]: Typo I think. =p

Comment (maxim-belkin) [inline Formula/ejdb.rb]: 😜

File: Formula/ejdb.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
+# ejdb: Build a bottle for Linuxbrew
```
Comment (sjackman) [inline Formula/ejdb.rb]: Delete this line.

File: Formula/ejdb.rb (hunk 1 above)
Comment (jonchang) [inline Formula/ejdb.rb]: Late nights lead to typos 🤔 Will merge when tests are green.

**Comment by jonchang** [comment]
`brew test` failed because Circle is using gcc-4.8 instead of gcc-5 to compile.

**Comment by sjackman** [comment]
That's by design of `Homebrew/brew` that `brew test` uses the default compiler of the host system and uses `--env=std` rather than `--env=super`. I opened a PR to use the same compiler for `test` as was used to compile the formula, but it was rejected.  The discussion is over at https://github.com/Homebrew/brew/pull/33
I'm not sure what the fix is for the purposes of this formula to get the test to pass.

**Comment by jonchang** [comment]
Sounds like there won't be a general solution to this. I'm OK with just publishing the bottle and then pointing people to use gcc-5 instead of /usr/bin/gcc if it becomes an issue.

**Comment by sjackman** [comment]
The problem with `brew test` failing is that all of its dependencies will also fail on CI. For example when `bzip2` is updated, its CI will fail because `brew test ejdb` fails.

**Comment by sjackman** [comment]
```
test.c: In function ‘main’:
test.c:35:5: error: ‘for’ loop initial declarations are only allowed in C99 mode
     for (int i = 0 i < TCLISTNUM(res); ++i) {
     ^
test.c:35:5: note: use option -std=c99 or -std=gnu99 to compile your code
Error: ejdb: failed
```
The test failure is easy to fix in this case. Either add `-std=c99` as the compiler suggests, or change
```c
for (int i = 0;
```
to
```c
int ;
for (i = 0;
```

**Comment by jonchang** [comment]
I pushed the fix, but should it be upstreamed to homebrew/core?

**Comment by jonchang** [comment]
Linker errors now. I'll try to figure this out tonight.

```
==> /usr/bin/gcc-4.8 -I/home/linuxbrew/.linuxbrew/Cellar/ejdb/1.2.12/include -L/home/linuxbrew/.linuxbrew/Cellar/ejdb/1.2.12/lib -lejdb test.c -o /tmp/ejdb-test-20170322-51558-w0t81x/test
.
Last 15 lines from /home/ubuntu/.cache/Homebrew/Logs/ejdb/test.01.gcc-4.8:
test.c:(.text+0x102): undefined reference to `bson_destroy'
test.c:(.text+0x111): undefined reference to `bson_init_as_query'
test.c:(.text+0x125): undefined reference to `bson_append_start_object'
test.c:(.text+0x13e): undefined reference to `bson_append_string'
test.c:(.text+0x14d): undefined reference to `bson_append_finish_object'
test.c:(.text+0x15c): undefined reference to `bson_finish'
test.c:(.text+0x182): undefined reference to `ejdbcreatequery'
test.c:(.text+0x1b1): undefined reference to `ejdbqryexecute'
test.c:(.text+0x207): undefined reference to `bson_print_raw'
test.c:(.text+0x22f): undefined reference to `tclistdel'
test.c:(.text+0x23e): undefined reference to `ejdbquerydel'
test.c:(.text+0x24d): undefined reference to `bson_destroy'
test.c:(.text+0x25c): undefined reference to `ejdbclose'
test.c:(.text+0x26b): undefined reference to `ejdbdel'
collect2: error: ld returned 1 exit status
```

**Comment by sjackman** [comment]
> I pushed the fix, but should it be upstreamed to homebrew/core?

Yes, I'd say so. It's possible that older versions of Mac (like perhaps Tiger) have a similar issue.

**Comment by sjackman** [comment]
This issue comes up often.
```ruby
system ENV.cc, "-I#{include}", "-L#{lib}", "-lejdb", "test.c", "-o", testpath/"test"
```
is non-portable and should be
```ruby
system ENV.cc, "-I#{include}", "test.c", "-L#{lib}", "-lejdb", "-o", testpath/"test"
```
The `-l` options should come after the source file.

**Comment by sjackman** [comment]
🍏 💚

**Pull request closed without merging**